File: 20240601070 SE226 LAB1/LAB2/lab2.cpp
#include <iostream>
using namespace std;
int main() {
    int num{};
    int sum{0};
    int temp{};
    std::cout << "Please enter a positive integer greater than 9: ";
    std::cin >> num;
    while (num <= 9) {
        std::cout << "Invalid input. Please enter a positive integer greater than 9: ";
        std:: cin >> num;
    }
    temp = num;
    std::cout << temp;

    while (temp > 9) {
        int sum = 0;
        while (temp > 0) {
            sum += temp % 10;
            temp /= 10;
        }
        temp = sum;
        std::cout << " -> " << sum;
    }
    std::cout << std::endl;




    // Task 2
    int val{};
    int count{1};
    std::cout << "Please enter a positive integer between 1 and 100: ";
    std::cin >> val;
    while (val < 10 || val > 100) {
        std::cout << "Invalid input. Please enter a positive integer between 1 and 100: ";
        std::cin >> val;
    }
    int fizzSteps{0};
    int buzzSteps{0};
    int fizzBuzzSteps{0};

    while (count <= val) {
        if (count % 7 == 0) {
            std::cout << "(" << count << " is skipped)" << std::endl;
            count++;
            continue;
        }
        if (count % 3 == 0) {
            if (count % 5 == 0) {
                std::cout << "FizzBuzz" << std::endl;
                fizzBuzzSteps++;
            } else {
                std::cout << "Fizz" << std::endl;
                fizzSteps++;
            }
        } else if (count % 5 == 0) {
            std::cout << "Buzz" << std::endl;
            buzzSteps++;
        } else {
            std::cout << count << std::endl;
        }
        count++;

    }
    std::cout << "----- Summary -----"<< std::endl;
    std::cout << "Fizz count: " << fizzSteps << std::endl;
    std::cout << "Buzz count: " << buzzSteps << std::endl;
    std::cout << "FizzBuzz count: " << fizzBuzzSteps << std::endl;




    // Task 3
    int pat{0};
    std::cout << "Please enter a number between 3 and 9:";
    std::cin >> pat;
    while (pat < 3 || pat > 9) {
        std::cout << "Invalid input. Please enter a number between 3 and 9:";
    }
    int i{1};
    while (i <= 2*pat-1) {
        int k = pat - abs(pat -i);
        for (int j = 1; j <= k; j++) {
            std::cout << j;
        }
        std::cout << std::endl;
        i++;
        if (i <1) {
            break;
        }
    }

    return 0;
}

File: LAB2/lab2.py
# Task 1
dsmvalue = None
dsmtemp = None

print("Please enter a positive integer greater than 9:")
dsmvalue = int(input())
while dsmvalue <= 9:
    print("Invalid input.Please enter a positive integer greater than 9:")
    dsmvalue = int(input())

dsmtemp = dsmvalue
print(str(dsmtemp), end=" ")
while dsmtemp > 9:
    dsmsum = 0
    while dsmtemp > 0:
        dsmsum = dsmsum + dsmtemp % 10
        dsmtemp = dsmtemp // 10

    dsmtemp = dsmsum
    print("-> " + str(dsmsum), end=" ")

print()

# Task 2
fbvalue = None
fbcount = 1
print("Please enter a positive integer between 1 and 100:")
fbvalue = int(input())
while fbvalue < 1 or fbvalue > 100:
    print("Invalid input. Please enter a positive integer between 1 and 100:")
    fbvalue = int(input())

fizzSteps = 0
buzzSteps = 0
fizzBuzzSteps = 0

while fbcount <= fbvalue:
    if fbcount % 7 == 0:
        print("(" + str(fbcount) + " is skipped)")
        fbcount+=1
        continue

    if fbcount % 3 == 0:
        if fbcount % 5 == 0:
            print("FizzBuzz")
            fizzBuzzSteps += 1
        else:
            print("Fizz")
            fizzSteps += 1
    elif fbcount % 5 == 0:
        print("Buzz")
        buzzSteps += 1
    else:
        print(fbcount)

    fbcount+=1

print("----- Summary -----")
print("Fizz count: " + str(fizzSteps))
print("Buzz count: " + str(buzzSteps))
print("FizzBuzz count: " + str(fizzBuzzSteps))

print()


# Task 3

pat = None
print("Please enter a number between 3 and 9:")
pat = int(input())
while pat < 3 or pat > 9:
    print("Invalid input. Please enter a number between 3 and 9:")
    pat = int(input())

i = 1
while i <= 2*pat-1:
    k = pat - abs(pat-i)
    for j in range(1, k+1):
        print(j, end="")
    print()
    i+=1
    if i < 1:
        break


